hail/src/main/scala/is/hail/expr/ir/EmitFunctionBuilder.scala

```diff
@@ -338,6 +362,9 @@ class EmitFunctionBuilder[F >: Null](
     val n = name.replace("/",".")
     val localHConf = _hconf
 
+    val useSpark = _sparkField != null
+    val spark = if (useSpark) new SparkBackendUtils(_mods.result()) else null
```

Comment:
This is kind of weird. My guess as to why this is happening is so that you generate the `SparkBackendUtils` on the master? Unfortunate that it can be set to null and then have to check `useSpark` twice.

Reply:
yeah, pretty much. And to avoid serializing stuff if it doesn't exist----this should only be non-null currently if the function's going to be called on the master node, since we can't start distributed jobs on worker nodes, and we don't want to serialize a bunch of empty classes that aren't going to be used.